ExampleMod/Items/Weapons/ExampleMagicMissile.cs

```diff
@@ -31,16 +32,30 @@ public class ExampleMagicMissile : ModItem
 		}
 
 		// This item's mana usage changes through the day, peaking at 1.5x mana usage at noon, and 0.5x mana usage at midnight.
-		public override void GetManaCost(Player player, ref int mana) {
+		public override void ModifyManaCost(Player player, ref float reduce, ref float mult) {
 			float currentTime = (float)Main.time;
+			// The time at which it changes from day to night and vice versa.
 			int maxTime = Main.dayTime ? 54000 : 32400;
+			// The time at which it is 12pm or 12am, which isn't exactly half the values above.
 			int time12 = Main.dayTime ? 28800 : 18000;
+			// If the time it after 12, recalculate the current time to account for the length of the second half of the day instead of the first.
 			if (currentTime > time12) {
+				currentTime -= time12;
 				time12 = maxTime - time12;
-				currentTime = maxTime - currentTime;
+				currentTime += time12;
 			}
-			float multi = 1 + currentTime / time12 * 0.5f * Main.dayTime.ToDirectionInt();
-			mana = (int)(mana * multi);
+			// If it is night time, add twice the duration of the current half of the night.
+			if (!Main.dayTime) {
+				currentTime += time12 * 2;
+			}
+			// We divide the current time by the current half of the day or night's time, then by 2.
+			// The result would gradually go from 0 at 4:30am, 0.5 at 12pm, 1 at 7:30pm and 1.5f at 12 am.
+			float timeMult = currentTime / time12 / 2;
+			// Then we multiply the result above by PI, and calculate the sine of that, then multiply it by 0.5f and add 1.
+			// The result would go from 1 at 4:30am, 1.5 at 12pm, 1 at 7:30pm and 0.5f at 12 am, in a sine wave pattern.
+			timeMult = 1 + (float)Math.Sin(timeMult * Math.PI) * 0.5f;
```

Comment:
```
// goal, 1.5x mana usage at noon, 0.5 at midnight
float time = Main.time;
if (!Main.dayTime) time += 54000; //Main.time resets at night, so add the length of the day (15 hours)
// 86400 ticks (ingame minutes, real seconds) per day. 3600 ticks per ingame hour
// day starts at 4:30am so add the time between midnight and 4:30
time += 16200; //time is now measured as time since midnight
// period of sin is 2pi, period of day is 86400 ticks
// cos peaks at 0, but peak mana usage should be at midday. Either shift 12 hours, or invert (we invert here)
// cos goes between -1 and 1, so multiply by 0.5 and add 0.5 to make it go between 0.5 and 1.5
double timeMult = 0.5 - 0.5 * Math.Cos(time / 86400 * 2 * Math.PI);
```

Reply:
Neither of these really do what i wanted to to show: 
* Normal mana usage at dawn / dusk, which are at 4:30 am and 7:30 pm in Terraria for some reason.
* And, coupled with the above, 1.5x mana usage at noon, 0.5 at midnight.

Because 12pm isn't the midpoint between dawn and dusk, nor is 12am the midpoint between dusk and dawn, each 4th of a day period differs in length.

* Your first example only matches the mana usage multiplier at noon and midnight, then it becomes normal at 6am/pm. This would match what would be expected but i wanted to show exactly when it changes between day and night and vice versa.
* Your second example only matches the mana usage multiplier at dusk and dawn, but differs at noon and midnight.

Reply:
Looks like I did miss that, but 4.5+7.5 = 12
Are you sure midnight isn't right inbetween 4:30 and 7:30 (and hence midday too)

Reply:
What i mean is that 12 / 2 == 6, but 4.5+6 != 12, so mana usage would peak at 10:30 in the day, and at 1:30 at night. It would be a different story if dawn and dusk were at 6.

Reply:
I think you'll find the second example works. Neither 12, 6 or 2 are used. Just the length of day/night becomes 0-pi in the sin

Reply:
"Because 12pm isn't the midpoint between dawn and dusk, nor is 12am the midpoint between dusk and dawn, each 4th of a day period differs in length."

I think you might be wrong here. 12 is halfway between 4.5 and 17.5, likewise there are 4.5 hours between 7:30 and midnight, and 4.5 hours between midnight and dawn

Reply:
Damn, you are completely right, i got so confused with the day's length, i somehow thought that the time between dawn and noon was different to the time from non to dusk, same for night time.